presto-main/src/main/java/com/facebook/presto/operator/scalar/ArraySortFunction.java

```diff
@@ -115,6 +117,56 @@ public int compare(Integer p1, Integer p2)
         return blockBuilder.build();
     }
 
+    @SqlType("array(bigint)")
+    public static Block bigintSort(@SqlType("array(bigint)") Block array)
+    {
+        final int arrayLength = array.getPositionCount();
+        if (arrayLength < 2) {
+            return array;
+        }
+
+        long[] values = new long[array.getPositionCount()];
+        boolean[] nulls = new boolean[array.getPositionCount()];
+
+        if (array.mayHaveNull()) {
+            for (int i = 0; i < array.getPositionCount(); i++) {
+                if (array.isNull(i)) {
+                    values[i] = Long.MAX_VALUE;
+                    nulls[i] = true;
+                }
+                else {
+                    values[i] = array.getLong(i);
+                }
+            }
+        }
+        else {
+            for (int i = 0; i < array.getPositionCount(); i++) {
+                values[i] = array.getLong(i);
+            }
+        }
+
+        Arrays.sort(values);
+
+        BlockBuilder blockBuilder = BIGINT.createBlockBuilder(null, arrayLength);
+        if (array.mayHaveNull()) {
```

Comment:
Can you set a flag on L135 instead of the "maybe" part -- I think we want to be more sure on the output. Further, here you should check for NOT having nulls.

Reply:
This is a standard pattern for handling nulls efficiently. Look at other array operators

Reply:
Ok, but I am wondering if you should be using the negation of this condition: the loop in 152 assumes no nulls, so maybe !array.maybeHaveNull() ?